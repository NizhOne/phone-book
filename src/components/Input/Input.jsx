import React from 'react';
import PropTypes from 'prop-types';

const Input = ({type, name, onChange, placeholder, value}) =>
    <label className={`label-input`}>
        <input type={type}
               name={name}
               placeholder={placeholder}
               value={value}
               onChange={onChange}
        />
    </label>;

Input.propTypes = {
    type: PropTypes.string,
    name: PropTypes.string,
    onChange: PropTypes.func.isRequired,
    placeholder: PropTypes.string,
    value: PropTypes.string.isRequired
};

export default Input;