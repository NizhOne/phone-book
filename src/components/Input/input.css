.label-input, .label-input input {
    width: 100%;
}