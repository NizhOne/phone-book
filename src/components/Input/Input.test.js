import React from 'react';
import Input from './Input';
import renderer from 'react-test-renderer';


test('test', () => {
    const component = renderer.create(
        <Input onChange={() => true} value={'hello'} />
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});