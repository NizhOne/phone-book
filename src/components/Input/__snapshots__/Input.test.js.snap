// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test 1`] = `
<label
  className="label-input"
>
  <input
    name={undefined}
    onChange={[Function]}
    placeholder={undefined}
    type={undefined}
    value="hello"
  />
</label>
`;
